beacon-chain/p2p/testing/p2p.go

```diff
@@ -47,6 +52,60 @@ func NewTestP2P(t *testing.T) *TestP2P {
 	}
 }
 
+// NewTestP2PWithKey initializes a new p2p test service.
+func NewTestP2PWithKey(t *testing.T, privkey crypto.PrivKey) *TestP2P {
+	ctx := context.Background()
+	h := bhost.NewBlankHost(GenSwarmWithKey(ctx, t, privkey))
```

Comment:
The only difference between this setup and the other one is the key, can you consolidate most of the logic to one? Maybe some inner function that takes the arguments. The way it is now, if we want to add an option to pubsub, we have to add it to both

Reply:
Removed